flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsResultPartition.java

```diff
@@ -186,17 +193,18 @@ public ResultSubpartitionView createSubpartitionView(
             throw new PartitionNotFoundException(getPartitionId());
         }
 
+        // assign a unique id for each consumer, now it is guaranteed by the value that is one
+        // higher than the last consumerId's id field.
+        HsConsumerId consumerId = nextConsumerIds[subpartitionId];
+        nextConsumerIds[subpartitionId] = new HsConsumerId(consumerId.getId() + 1);
```

Comment:
See my other comments for consumer id generation and multiple consumer checking.